src/main/java/org/folio/dew/utils/QueryUtils.java

```diff
@@ -0,0 +1,119 @@
+      expressions[expressions.length - 1] = matcher.group(1);
+      sorting = matcher.group(2);
+    }
+
+    var suffix = CQL_SUFFIX + sorting;
+    var delimiter = String.format(CQL_COMBINE_OPERATOR, operator);
+    return StreamEx.of(expressions)
+      .filter(StringUtils::isNotBlank)
+      .joining(delimiter, CQL_PREFIX, suffix);
+  }
+
+  /**
+   * Converts a collection of IDs to a CQL query string using the specified ID field.
+   *
+   * @param ids     The collection of IDs to be converted.
+   * @param idField The field name to be used in the CQL query.
+   * @return A CQL query string representing the IDs.
+   */
+  public static String convertIdsToCqlQuery(Collection<String> ids, String idField) {
+    return convertFieldListToCqlQuery(ids, idField, true, false);
+  }
+
+  /**
+   * Converts a collection of IDs to a CQL query string using the default ID field.
+   *
+   * @param ids The collection of IDs to be converted.
+   * @return A CQL query string representing the IDs.
+   */
+  public static String convertIdsToCqlQuery(Collection<String> ids) {
+    return convertFieldListToCqlQuery(ids, ID, true, false);
+  }
+
+  /**
+   * Transform list of values for some property to CQL query using 'or' operation and enclosing values with quotes
+   *
+   * @param values      list of field values
+   * @param fieldName   the property name to search by
+   * @param strictMatch indicates whether strict match mode (i.e. ==) should be used or not (i.e. =)
+   * @return String representing CQL query to get records by some property enclosed values
+   */
+  public static String convertFieldListToEnclosedCqlQuery(Collection<?> values, String fieldName, boolean strictMatch) {
+    return convertFieldListToCqlQuery(values, fieldName, strictMatch, true);
+  }
+
+  /**
+   * Transform list of values for some property to CQL query using 'or' operation
+   *
+   * @param values      list of field values
+   * @param fieldName   the property name to search by
+   * @param strictMatch indicates whether strict match mode (i.e. ==) should be used or not (i.e. =)
+   * @param enclosed    indicates whether values should be enclosed with quotes (i.e. asd) or not (i.e. "asd")
+   * @return String representing CQL query to get records by some property values
+   */
+  public static String convertFieldListToCqlQuery(Collection<?> values, String fieldName, boolean strictMatch, boolean enclosed) {
+    var prefix = String.format(strictMatch ? CQL_MATCH_STRICT : CQL_MATCH, fieldName, CQL_PREFIX);
+    var enclose = enclosed ? "\"%s\"" : "%s";
+    return StreamEx.of(values)
+      .map(Object::toString)
+      .map(enclose::formatted)
+      .joining(" or ", prefix, CQL_SUFFIX);
+  }
+
+  public static String negateQuery(String cql) {
+    return CQL_NEGATE_PREFIX + cql;
+  }
+
+  public static String getCqlExpressionForFieldNullValue(String fieldName) {
+    return String.format(CQL_UNDEFINED_FIELD_EXPRESSION, fieldName);
+  }
+
+  private QueryUtils() {}
```

Comment:
we usually have private constructor at the beggining of utility class in other places

Reply:
Moved it to the top